kafka/4.2/modules/ROOT/pages/kafka-connector-reference.adoc

```diff
@@ -0,0 +1,1558 @@
+** DAYS |  Determines the time unit for the delivery timeout scalar |  SECONDS |
+| Enable idempotence a| Boolean |  When set to `true`, the producer ensures that exactly one copy of each message is written in the stream. If set to `false`, the producer retries due to broker failures and other conditions can write duplicates of the retried message in the stream. Enabling idempotence requires the value of `max.in.flight.requests.per.connection` to be less than or equal to `5`, retries to be greater than `0`, and acks to be `all`. If you do not explicitly set these values, suitable values are chosen. If incompatible values are set, a ConnectionException is thrown. |  false |
+| Linger time  a| Number |  The producer groups records that arrive in between request transmissions into a single batched request. Normally, this occurs under load, when records arrive faster than they can be sent out. However, in some circumstances, the client might want to reduce the number of requests, even under a moderate load.
+*Reviewers, should "the client" in the previous sentence be changed to "the producer"?*
+{sp} +
+{sp} +
+This setting adds a small amount of artificial delay to the sending of recrords. Rather than immediately sending out a record, the producer waits for up to the specified delay to allow other records to be sent so it can batch the records together. This is analogous to Nagle's algorithm in TCP. The setting gives the upper bound on the delay for batching. Once the connector receives `batch.size` worth of records for a partition, it sends the batch immediately, regardless of this setting.
+{sp} +
+{sp} +
+This setting defaults to `0`, which means no delay. Setting `linger.ms=5`, for example, has the same effect of reducing the number of requests sent, but adds up to 5ms of latency to records sent in the absence of load. |  0 |
+| Linger Time Unit a| Enumeration, one of:
+
+** NANOSECONDS
+** MICROSECONDS
+** MILLISECONDS
+** SECONDS
+** MINUTES
+** HOURS
+** DAYS |  The time unit for the linger time scalar |  SECONDS |
+| Maximum block time  a| Number |  Specifies for how long `KafkaProducer.send()` and `KafkaProducer.partitionsFor()` can be blocked. These methods can be blocked either because the buffer is full or because the `unavailable.Blocking` metadata in the user-supplied serializers or partitioner are counted against this timeout. |  60 |
+| Maximum block time unit a| Enumeration, one of:
+
+** NANOSECONDS
+** MICROSECONDS
+** MILLISECONDS
+** SECONDS
+** MINUTES
+** HOURS
+** DAYS |  The time unit for the maximum block time scalar |  SECONDS |
+| Maximum in flight requests a| Number |  The maximum number of unacknowledged requests the client sends on a single connection before blocking. If this setting is greater than `1` and there are failed sends, there is a risk of message re-ordering due to retries (for example, if retries are enabled). *Reviewers, should "the client" in the previous sentence be changed to "the producer"?*|  5 |
```

Comment:
```suggestion
| Maximum in flight requests a| Number |  The maximum number of unacknowledged requests the client sends on a single connection before blocking. If this setting is greater than `1` and there are failed sends, there is a risk of message re-ordering due to retries if retries are enabled. |  5 |
```

Reply:
It looks like you are embedding reviewer questions in the file.  ? Don't forget to remove these.

Reply:
Producer = client